addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -190,13 +190,19 @@ def _get_bom_data(self, bom, warehouse, product=False, line_qty=False, bom_line=
         has_attachments = False
         if not is_minimized:
             if product:
-                has_attachments = self.env['product.document'].search_count(['&', '&', ('attached_on_mrp', '=', 'bom'), ('active', '=', True), '|', '&', ('res_model', '=', 'product.product'),
-                                                                 ('res_id', '=', product.id), '&', ('res_model', '=', 'product.template'),
-                                                                 ('res_id', '=', product.product_tmpl_id.id)], limit=1) > 0
+                has_attachments = self.env['product.document'].search_count([
+                    '&', ('res_field', '=', False),
+                    '&', '&', ('attached_on_mrp', '=', 'bom'), ('active', '=', True), '|',
+                    '&', ('res_model', '=', 'product.product'), ('res_id', '=', product.id),
```

Comment:
Side note: we could simplify those domains by allowing recordsets on the right-hand-side of conditions.  In the case of many2one_reference fields, this would use the record's model to constraint the corresponding model field:
```py
assert field.type == 'many2one_reference'
Domain(field.name, 'in', records)
   =>   Domain(field.model_field, '=', records._name) & Domain(field.name, 'in', records.ids)
```

Reply:
I have a small POC on the branch `master-domain-value-model-krma`.